lighthouse-core/report/templates/report-template.html

```diff
@@ -247,8 +249,14 @@ <h2 class="config-section__title">Blocked URL Patterns</h2>
 {{/each}}
 
 <script>
-window.addEventListener('DOMContentLoaded', _ => {
-  new LighthouseReport(window.lhresults);
+(function(exports) {
```

Comment:
Is the confusion mostly about the inline `<script>` at the bottom of report-template.html (@paulirish's 2nd screenshot)? I can create a separate .js file with the `DOMContentLoaded` setup  and load it in the `getReportJS()` array. Would that help?

> why is report-generator including the source of scripts/lighthouse-report into the report... 

report-generator.js has always included the ligthhouse-report.js source into the report. I didn't change that setup. The only difference in this PR is that we're inlining more JS into the report (logger.js + filer-namer.js + lighthouse-report.js).

> ....when that file was already part of the bundle that shipped with the viewer

Viewer bundles report-generator.js **as-is** because it needs all the mechanics that generate the HTML report. Viewer needs all of report-generator.js because it dynamically generates an HTML report  when the user drops a json file, pastes a link, or deep links from github. The idea was that we have the same codepath (`ReportGenerator.generateHTML()`) generating HTML for all three: crx/cli/viewer.

> also, can we rename some of these files, because they all sound like the same thing right now. :)

Partially why I'm advocating we don't create more micro files :) lighthouse-report.js and the viewer's subclass (lighthouse-report-viewer.js) make sense to me, but I'm open to renaming.